Second order ds

vacary opened the pull request:
The object of this PR is to add a SecondOrderDS class as a base class for NewtonEulerDS and LagrangianDS, It should:

- help to mix both class in examples
- factorize some code in OSI and OSNS

vacary committed:
[kernel] start to implement SeconsOrderDS as a base class for LagrangianDS and NewtonEulerDS

vacary committed:
[kernel] come back to SiconosMatrix for _mass

xhub reviewed (changes requested):
I like the idea. I just noticed some potential issues with virtual (or-non virtual) methods

vacary committed:
[kernel] start to simplify MoreauJeanOSI with SecondOrderDS

vacary committed:
[kernel] add virtual methods in SecondOrderDS

vacary committed:
[kernel] add missing file

vacary committed:
[kernel] come back to SiconosMatrix for constructors of LagrangianLinearTIDS

vacary committed:
Merge branch 'master' into SecondOrderDS

vacary committed:
Merge branch 'master' into SecondOrderDS

vacary committed:
Merge branch 'master' into SecondOrderDS

vacary committed:
[kernel] remove from directors the methods that return a pointer of a reference.

	 Following the warning of swig :
	 Warning(473): Returning a pointer or reference in a director method is not recommended.
	 I remove the methods that return an attribute from the director.
	 see a discussion here https://sourceforge.net/p/swig/mailman/message/7200412/

	 Take a look at  the C++ code generated. You'll see that the return value
	 passed to C++ code points to a Python/Java object. There is no guarantee
	 as to the lifetime of that object. If you are sure the object returned
	 will not be garbage collected while the calling function is using it you
	 should be okay. Be careful as gc happens in a second thread in Java, not
	 sure about Python or Ruby. This is just a limitation of working with the
	 target language and using pointers/references and is not easily solved.
	 For example, you could modify the directorout typemap to take a copy of
	 the object, but then you most likely have a memory leak on your hands or
	 if you use a local static variable to copy the object, you'll end up
	 with a thread safety issue.
